tools/nni_cmd/launcher.py

```diff
@@ -553,7 +560,15 @@ def manage_stopped_experiment(args, mode):
     new_config_file_name = ''.join(random.sample(string.ascii_letters + string.digits, 8))
     new_nni_config = Config(new_config_file_name)
     new_nni_config.set_config('experimentConfig', experiment_config)
-    launch_experiment(args, experiment_config, mode, new_config_file_name, experiment_id)
+    try:
+        launch_experiment(args, experiment_config, mode, new_config_file_name, experiment_id)
+    except Exception as exception:
+        nni_config = Config(new_config_file_name)
+        restServerPid = nni_config.get_config('restServerPid')
+        if restServerPid:
+            kill_command(restServerPid)
+        print_error(exception)
+        exit(1)
     new_nni_config.set_config('restServerPort', args.port)
```

Comment:
curious about this line. if users use `--foreground`, this line would not be executed, thus, new rest port would not be set, is this a problem?

Reply:
good point, moved `new_nni_config.set_config('restServerPort', args.port)` before `launch_experiment()` function.